File: Rubik.HTML/Server/Schema/DimensionInfo.cs
using System;
using System.Web;
using System.ComponentModel;
using System.Collections.Generic;

namespace Rubik.Server.Schema
{
    public class DimensionInfo : InfoObject
    {
        public List<HierarchyInfo> Hierarchies
        {
            get; set;
        }

        public CubeInfo Cube
        {
            get; set;
        }
    }
}

File: Rubik.HTML/Lib/DataHub/Schema/DimensionInfo.ts
module Rubik.DataHub {
    export class DimensionInfo extends InfoObject {
        get ObjectType(): ObjectTypeEnum { return ObjectTypeEnum.Dimension; }

        Cube_UniqueName: string;
        DefaultHierarchy: string;
        
        get Hierarchies(): Promise<HierarchyInfo[]> {
            return this.factory.GetInfoObjectCollection(Rubik.DataHub.ObjectTypeEnum.Hierarchy, this) as Promise<HierarchyInfo[]>;  
        }
        get Cube(): Promise<CubeInfo> {
            return this.factory.GetAncestorInfo(Rubik.DataHub.ObjectTypeEnum.Cube, this) as Promise<CubeInfo>;                    
        }
    }
}